src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -119,49 +124,137 @@ public function load($class, $type = null)
             }
         }
 
+        /** @var $annot RouteAnnotation */
         if (0 === $collection->count() && $class->hasMethod('__invoke') && $annot = $this->reader->getClassAnnotation($class, $this->routeAnnotationClass)) {
             $globals['path'] = '';
             $globals['name'] = '';
+            $globals['locales'] = array();
             $this->addRoute($collection, $annot, $globals, $class, $class->getMethod('__invoke'));
         }
 
         return $collection;
     }
 
-    protected function addRoute(RouteCollection $collection, $annot, $globals, \ReflectionClass $class, \ReflectionMethod $method)
+    /**
+     * @param RouteCollection  $collection
+     * @param RouteAnnotation  $annotation
+     * @param array            $globals
+     * @param ReflectionClass  $class
+     * @param ReflectionMethod $method
+     */
+    protected function addRoute(RouteCollection $collection, $annotation, $globals, ReflectionClass $class, ReflectionMethod $method)
     {
-        $name = $annot->getName();
+        $name = $annotation->getName();
```

Comment:
please don't rename the variable. It will make merging older branches into master harder for no benefit (and we will have to do it for years to come)

Reply:
sad we cannot go for better...

Reply:
well, renaming it only in master makes our maintenance work harder for the next 5 years, each time this code would be modified. That's clearly not good (especially given that this would only make this code better for contributors, as users will never see this variable name)